### PR TITLE
File 290

### PR DESCRIPTION
**Pull request by flyguest**
Having separate routes file for the AdminApp.

### CODE REVIEW

File: app/uk/gov/hmrc/fileupload/controllers/AdminController.scala [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import play.api.mvc.{Action, BodyParser, Controller, MultipartFormData}
+import uk.gov.hmrc.fileupload.fileupload._
+import uk.gov.hmrc.fileupload.notifier.NotifierService._
+import uk.gov.hmrc.fileupload.{EnvelopeId, FileId, FileRefId}
+import uk.gov.hmrc.fileupload.transfer.TransferRequested
+import uk.gov.hmrc.fileupload.virusscan.VirusScanRequested
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AdminController (uploadParser: () => BodyParser[MultipartFormData[Future[JSONReadFile]]],
```
Comment (hannesstockner) [inline app/uk/gov/hmrc/fileupload/controllers/AdminController.scala]: uploadParser is not needed here, isn't?

File: app/uk/gov/hmrc/fileupload/controllers/AdminController.scala [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                          notify: AnyRef => Future[NotifyResult],
+                          now: () => Long)
```
Comment (hannesstockner) [inline app/uk/gov/hmrc/fileupload/controllers/AdminController.scala]: now is not needed

File: app/uk/gov/hmrc/fileupload/controllers/AdminController.scala [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    Future.successful(Ok)
+  }
+
+  def transfer(envelopeId: EnvelopeId, fileId: FileId, fileRefId: FileRefId) = Action.async { request =>
+    notify(TransferRequested(envelopeId = envelopeId, fileId = fileId, fileRefId = fileRefId))
+    Future.successful(Ok)
+  }
+}
+
+object AdminController {
```
Comment (hannesstockner) [inline app/uk/gov/hmrc/fileupload/controllers/AdminController.scala]: is this needed here?

**Comment by flyguest** [comment]
Thanks for viewing.
Unnecessary codes have been removed.

File: app/uk/gov/hmrc/fileupload/controllers/AdminController.scala (hunk 1 above)
Comment (hannesstockner) [inline app/uk/gov/hmrc/fileupload/controllers/AdminController.scala]: uploadParser is not needed here

File: app/uk/gov/hmrc/fileupload/frontendGlobal.scala [hunk 4]
```diff
@@ -74,6 +74,7 @@ object FrontendGlobal
     Akka.system.actorOf(TransferActor.props(subscribe, streamTransferCall), "transferActor")
 
     fileUploadController
+    AdminController
```
Comment (hannesstockner) [inline app/uk/gov/hmrc/fileupload/frontendGlobal.scala]: variable names are always lower case

Comment (flyguest) [inline app/uk/gov/hmrc/fileupload/frontendGlobal.scala]: thanks again,
it has been changed

File: app/uk/gov/hmrc/fileupload/controllers/AdminController.scala [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import uk.gov.hmrc.fileupload.fileupload._
+import uk.gov.hmrc.fileupload.notifier.NotifierService._
+import uk.gov.hmrc.fileupload.{EnvelopeId, FileId, FileRefId}
+import uk.gov.hmrc.fileupload.transfer.TransferRequested
+import uk.gov.hmrc.fileupload.virusscan.VirusScanRequested
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class AdminController (notify: AnyRef => Future[NotifyResult])
+                         (implicit executionContext: ExecutionContext) extends Controller {
```
Comment (hannesstockner) [inline app/uk/gov/hmrc/fileupload/controllers/AdminController.scala]: not aligned

File: app/uk/gov/hmrc/fileupload/controllers/AdminController.scala [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+package uk.gov.hmrc.fileupload.controllers
+
+import play.api.libs.json.{JsObject, JsString, Json}
+import play.api.mvc.{Action, BodyParser, Controller, MultipartFormData}
+import uk.gov.hmrc.fileupload.fileupload._
+import uk.gov.hmrc.fileupload.notifier.NotifierService._
+import uk.gov.hmrc.fileupload.{EnvelopeId, FileId, FileRefId}
+import uk.gov.hmrc.fileupload.transfer.TransferRequested
+import uk.gov.hmrc.fileupload.virusscan.VirusScanRequested
+
```
Comment (hannesstockner) [inline app/uk/gov/hmrc/fileupload/controllers/AdminController.scala]: I guess not all the importans are needed anymore

Comment (flyguest) [inline app/uk/gov/hmrc/fileupload/controllers/AdminController.scala]: they have been fixed

**Pull request merged**